app/components/SongCard/index.js

```
@@ -0,0 +1,101 @@
+    } else {
+      audioRef.current.play();
+    }
+    setIsPlaying(!isPlaying);
+  };
+
+  React.useEffect(() => {
+    return () => {
+      if (audioRef.current) {
+        audioRef.current.pause();
+        audioRef.current.removeEventListener('ended', () => setIsPlaying(false));
+        audioRef.current = null;
```

ESLint

[error] No object mutation allowed.

(immutable/no-mutation)